php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1532 @@
+      .is(f.dimensionalOffset(
+        b.token(LBRACKET),
+        b.optional(EXPRESSION()),
+        b.token(RBRACKET)
+      ));
+  }
+
+  public ExpressionTree PRIMARY_EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.PRIMARY_EXPRESSION)
+      .is(
+        b.firstOf(
+          f.newStaticIdentifier(b.token(STATIC)),
+          // FIXME (Linda): add b.sequence(PHPKeyword.NAMESPACE, FULLY_QUALIFIED_CLASS_NAME)
+          VARIABLE_WITHOUT_OBJECTS(),
+          IDENTIFIER(),
+          PARENTHESIZED_EXPRESSION()
+        ));
+  }
+
+  public AssignmentByReferenceTree ASSIGNMENT_BY_REFERENCE() {
+    return b.<AssignmentByReferenceTree>nonterminal(PHPLexicalGrammar.ASSIGNMENT_BY_REFERENCE).is(
+      f.assignmentByReference(
+        MEMBER_EXPRESSION(),
+        b.token(PHPPunctuator.EQU), b.token(AMPERSAND),
+        b.firstOf(NEW_EXPRESSION(), MEMBER_EXPRESSION())));
+  }
+
+  public AssignmentExpressionTree ASSIGNMENT_EXPRESSION() {
+    return b.<AssignmentExpressionTree>nonterminal(PHPLexicalGrammar.ASSIGNMENT_EXPRESSION).is(
+      f.assignmentExpression(MEMBER_EXPRESSION(), ASSIGNMENT_OPERATOR(), EXPRESSION()));
+  }
+
+  public InternalSyntaxToken ASSIGNMENT_OPERATOR() {
+    return b.<InternalSyntaxToken>nonterminal(PHPLexicalGrammar.ASSIGNMENT_OPERATOR).is(
+      b.firstOf(
+        b.token(PHPPunctuator.EQU),
+
+        b.token(PHPPunctuator.STAR_EQU),
+        b.token(PHPPunctuator.DIVEQUAL),
+        b.token(PHPPunctuator.MOD_EQU),
+        b.token(PHPPunctuator.PLUS_EQU),
+        b.token(PHPPunctuator.MINUS_EQU),
+        b.token(PHPPunctuator.SL_EQU),
+        b.token(PHPPunctuator.SR_EQU),
+        b.token(PHPPunctuator.CONCATEQUAL),
+
+        b.token(PHPPunctuator.ANDEQUAL),
+        b.token(PHPPunctuator.XOR_EQU),
+        b.token(PHPPunctuator.OR_EQU)
+      ));
+  }
+
+  public ReferenceVariableTree REFERENCE_VARIABLE() {
+    return b.<ReferenceVariableTree>nonterminal(Kind.REFERENCE_VARIABLE)
+      .is(f.referenceVariable(
+        b.token(AMPERSAND),
+        MEMBER_EXPRESSION()));
+  }
+
+  public SpreadArgumentTree SPREAD_ARGUMENT() {
+    return b.<SpreadArgumentTree>nonterminal(Kind.SPREAD_ARGUMENT)
+      .is(f.spreadArgument(
+        b.token(ELIPSIS),
+        EXPRESSION()
+      ));
+  }
+
+  public FunctionCallTree FUNCTION_CALL_ARGUMENT_LIST() {
+    return b.<FunctionCallTree>nonterminal(PHPLexicalGrammar.FUNCTION_CALL_PARAMETER_LIST)
+      .is(f.functionCallParameterList(
+        b.token(LPARENTHESIS),
+        b.optional(f.newTuple9(
+          FUNCTION_CALL_ARGUMENT(), b.zeroOrMore(f.newTuple5(b.token(COMMA), FUNCTION_CALL_ARGUMENT())))),
+        b.token(RPARENTHESIS)));
+  }
+
+  public ExpressionTree FUNCTION_CALL_ARGUMENT() {
+    return b.<ExpressionTree>nonterminal()
+      .is(
+        b.firstOf(
+          REFERENCE_VARIABLE(),
+          SPREAD_ARGUMENT(),
+          EXPRESSION(),
+          YIELD_EXPRESSION()));
+  }
+
+  public ExpressionTree MEMBER_EXPRESSION() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)